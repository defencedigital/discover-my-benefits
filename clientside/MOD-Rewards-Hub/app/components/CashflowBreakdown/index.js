import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';

import CopyCol from '../CopyCol';
import ContainerInner from '../ContainerInner';

import { formatCurrency } from '../../utils/currency';

const CashflowBreakdown = props => (
  <div className="cashflow-breakdown">
    <ContainerInner>
      <Row>
        <Col xs="12" md="9">
          <h3 className="h2 cashflow-breakdown-title">{props.title}:</h3>
        </Col>
        <Col xs="12" md="3">
          <span className="cashflow-breakdown-title-figure">{formatCurrency(props.total)}</span>
        </Col>
      </Row>
      <Row>
        <Col xs="12">
          <CopyCol content={props.content} />
          <h4>{props.titleHint}</h4>
        </Col>
      </Row>
      <Row>
        <Col xs="12" md="9">
          <p className="cashflow-breakdown-txt">Pay back on {props.firstMonth}: </p>
        </Col>
        <Col xs="12" md="3">
          <span className="cashflow-breakdown-figure">{formatCurrency(props.monthly)}</span>
        </Col>
      </Row>
      <hr className="cashflow-breakdown-separator" />
      <Row>
        <Col xs="12" md="9">
          <p className="cashflow-breakdown-txt">Pay back on {props.secondMonth}: </p>
        </Col>
        <Col xs="12" md="3">
          <span className="cashflow-breakdown-figure">{formatCurrency(props.monthly)}</span>
        </Col>
      </Row>
      <hr className="cashflow-breakdown-separator" />
      <Row>
        <Col xs="12" md="9">
          <p className="cashflow-breakdown-txt">Pay back on {props.thirdMonth}: </p>
        </Col>
        <Col xs="12" md="3">
          <span className="cashflow-breakdown-figure">{formatCurrency(props.monthly)}</span>
        </Col>
      </Row>
    </ContainerInner>
  </div>
);

CashflowBreakdown.propTypes = {
  title: PropTypes.string,
  titleHint: PropTypes.string,
  content: PropTypes.string,
  total: PropTypes.string,
  firstMonth: PropTypes.string,
  secondMonth: PropTypes.string,
  thirdMonth: PropTypes.string,
  monthly: PropTypes.string,
};

export default CashflowBreakdown;
