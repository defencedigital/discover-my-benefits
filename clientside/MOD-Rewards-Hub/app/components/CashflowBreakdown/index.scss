.cashflow-breakdown {
  padding: 60px 0;
  margin: 50px 0 0;
  background-color: $brand-primary;
  position: relative;
  color: $text-color-invert;
}

.cashflow-breakdown-title {
  margin: 10px 0 40px 0;

  @media screen and (max-width: $breakpoint-sm) {
    margin: 10px 0 0;
  }
}

.cashflow-breakdown-separator {
  border-top: 1px solid $white;
  margin: 20px 0 20px;
  display: block;
  width: 100%;
}

.cashflow-breakdown-title-figure {
  padding: 0 20px;
  display: block;
  width: 100%;
  font-size: 2.5rem;
  font-weight: bold;
  text-align: center;
  margin: 0 auto;

  @media screen and (max-width: $breakpoint-sm) {
    text-align: left;
    padding: 0 0 20px;
  }
}

.cashflow-breakdown-figure {
  padding: 20px;
  display: block;
  width: 100%;
  font-size: 2.5rem;
  font-weight: bold;
  text-align: center;
  margin: 0 auto;

  @media screen and (max-width: $breakpoint-sm) {
    text-align: left;
    padding: 10px 0;
  }
}

.cashflow-breakdown-txt {
  padding: 40px 0 0;
  display: block;
  width: 100%;

  @media screen and (max-width: $breakpoint-sm) {
    padding: 10px 0 0;
  }
}

.cashflow-breakdown .copy-col {
  padding: 0 0 20px 0;
}

