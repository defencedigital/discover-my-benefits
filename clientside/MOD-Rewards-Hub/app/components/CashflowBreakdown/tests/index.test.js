import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CashflowBreakdown from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  titleHint: 'titleHint',
  content: 'content',
  total: 'total',
  firstMonth: 'firstMonth',
  secondMonth: 'secondMonth',
  thirdMonth: 'thirdMonth',
  monthly: 'monthly',
});

describe('CashflowBreakdown component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<CashflowBreakdown {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
