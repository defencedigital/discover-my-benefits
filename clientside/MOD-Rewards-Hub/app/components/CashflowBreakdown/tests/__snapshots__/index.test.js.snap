// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashflowBreakdown component matches the snapshot 1`] = `ShallowWrapper {}`;
